### ISSUE TITLE
Upgrade to openzeppelin upgrades plugin

### ISSUE THREAD

**Issue by nambrot** [celo-org/celo-monorepo#6433]
### Expected Behavior

Depend on `@openzeppelin/upgrades-core`

### Current Behavior

Depend on deprecated `@openzeppelin/upgrades`

**Comment by nambrot** [comment]
I feel like we should do this sooner than later, especially with eye to #6569 to catch potential catastrophic issues, wdyt @yorhodes @aslawson ?

**Comment by aslawson** [comment]
I'm not familiar with how this is related to #6569 ?

**Comment by nambrot** [comment]
#6599 specifies some work to ensure that core contracts are not vulnerable with these op codes.

As far as I know, the openzeppelin upgrades plugin can probably do this?

**Comment by yorhodes** [comment]
This consists of migrating our usage from [@openzeppelin/upgrades](https://www.npmjs.com/package/@openzeppelin/upgrades) to [@openzeppelin/upgrades-core](https://www.npmjs.com/package/@openzeppelin/upgrades-core) and [@openzeppelin/truffle-upgrades](https://www.npmjs.com/package/@openzeppelin/truffle-upgrades). We are consuming these tools in ways that the OpenZeppelin team does not intend, so there is major gymnastics involved in getting them to work with our suite including inlining unexported functionality and type wrangling.

I began this work on the branch `yorhodes/oz-upgrades` and spent at least 2 days working on the refactor. @nambrot suggested I reevaluate whether we want to do this as we had scoped for 1 day and the primary motivation was to accomplish #6569. There is probably 2 more days work involved in getting all of our tooling working and tests passing with this refactor, and I was fearful of making such large changes while lacking coverage.

This led me to briefly work on #6079 to have some better coverage for all of the historical version diffs we have deployed.

A much easier path to accomplishing #6569 will be documented within the ticket. For now I will close this as a won't fix and we can reopen if we have a need beyond #6569.
